File: src/main/java/network/adapter/UrlAdapter.java
package network.adapter;

import com.squareup.moshi.FromJson;
import com.squareup.moshi.ToJson;

import java.net.MalformedURLException;
import java.net.URL;

public class UrlAdapter {
    @ToJson String toJson(URL url) {
        return url.toString();
    }

    @FromJson URL fromJson(String json) {
        try {
            return new URL(json);
        } catch (MalformedURLException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: src/main/java/model/json/adapter/UrlAdapter.kt
package model.json.adapter

import com.squareup.moshi.FromJson
import com.squareup.moshi.ToJson

import java.net.MalformedURLException
import java.net.URL

class UrlAdapter {

    @ToJson
    internal fun toJson(url: URL): String {
        return url.toString()
    }

    @FromJson
    internal fun fromJson(json: String): URL? {
        try {
            return URL(json)
        } catch (e: MalformedURLException) {
            e.printStackTrace()
            return null
        }

    }

}
